e2e/playwright/search/src/tests/search-results-libraries.spec.ts

```diff
@@ -0,0 +1,237 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import {
+  ApiClientFactory,
+  Utils,
+  test,
+  TrashcanApi,
+  NodesApi,
+  SitesApi,
+  FileActionsApi,
+  SITE_VISIBILITY,
+  SITE_ROLES
+} from '@alfresco/playwright-shared';
+
+test.describe('Search Results - General', () => {
+  let trashcanApi: TrashcanApi;
+  let nodesApi: NodesApi;
+  let sitesApi: SitesApi;
+  let sitesAdminApi: SitesApi;
+  let fileActionsApi: FileActionsApi;
+
+  const username = `user-${Utils.random()}`;
+  const random = Utils.random();
+  const randomDescription = Utils.random() + '-' + Utils.random();
+
+  const site1 = {
+    name: `lib-${random}-1`,
+    id: `site-${random}-1`
+  };
+  const site2 = {
+    name: `site-2-${random}`,
+    id: `site-${random}-2`
+  };
+  const site3 = {
+    name: `lib-${random}-2`,
+    id: `site3-${random}`
+  };
+  const site4 = {
+    name: `my-site-${random}`,
+    id: `site4-${random}`,
+    description: randomDescription
+  };
+
+  const userSitePrivate = `user-site-${random}-private`;
+  const userSiteModerated = `user-site-${random}-moderated`;
+  const userSitePublic = `user-site-${random}-public`;
+
+  const adminSite1 = `admin-${random}-site1`;
+  const adminSite2 = `admin-${random}-site2`;
+  const adminSite3 = `admin-${random}-site3`;
+  const adminSite4 = `admin-${random}-site4`;
+  const adminPrivate = `admin-${random}-sitePrivate`;
+
+  const siteRussian = {
+    /* cspell:disable-next-line */
+    name: `любимый-сайт-${random}`,
+    id: `site-russian-id-${random}`
+  };
+
+  test.beforeEach(async ({ loginPage }) => {
+    await Utils.tryLoginUser(loginPage, username, username, 'beforeEach failed');
+  });
+
+  test.afterAll(async () => {
+    await Utils.deleteNodesSitesEmptyTrashcan(nodesApi, trashcanApi, 'afterAll failed', undefined, undefined, undefined);
```

Comment:
instead of passing 3 undefined you can pass nothing as these parameters are optional